samples/Webserver/WebServer.DI/TextServiceSingleton.cs

```diff
@@ -0,0 +1,25 @@
+﻿//
+// Copyright (c) 2020 Laurent Ellerbach and the project contributors
+// See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Text;
+
+namespace nanoFramework.WebServer.Sample
+{
+    internal class TextServiceSingleton : ITextServiceSingleton
```

Comment:
why do create 2 different classes which are exactly the same. Same for the interfaces. Only 1 is enough in both cases.
The difference is just when you inject it, either as a singleton either not. And I think it will be more clear as well

Reply:
The idea behind two same interfaces and classes is very simple - I just want to use both in the same controller and show that one is new one each time and second is the same across all requests. And there is no way to do it in DI, because if you register same interface two times only the second one will be resolved.

![image](https://user-images.githubusercontent.com/19893965/187675242-8a55b955-26b2-488b-be06-b98e513152b6.png)
![image](https://user-images.githubusercontent.com/19893965/187675268-c9f34b5e-46aa-4e87-88af-49d316b2a839.png)

Hope it's clear now why, two different interfaces are required :)


I've cleaned it a little bit, by removing "TextServiceSingleton" class.

Reply:
You can do it in a clear way as well with only 1 interface:

```csharp
public class ControllerTest
    {
        private readonly ITextService _textService;
        private readonly ITextService _textServiceSingleton;

        public ControllerTest(ITextService textService, ITextService textServiceSingleton)
        {
            _textService = textService;
            _textServiceSingleton = textServiceSingleton;
        }

        [Route("test")]
        [Method("GET")]
        public void RoutePostTest(WebServerEventArgs e)
        {
            var content = $"Response from {nameof(ITextService)}: {_textService.GetText()}. Response from {nameof(ITextServiceSingleton)}: {_textServiceSingleton.GetText()}";
            e.Context.Response.ContentType = "text/plain";
            WebServer.OutPutStream(e.Context.Response, content);
        }
    }
```

And also:

```csharp
private static ServiceProvider ConfigureServices()
        {
            return new ServiceCollection()
                .AddTransient(typeof(ITextService), typeof(TextService))
                .AddSingleton(typeof(ITextService), typeof(TextService))
                .BuildServiceProvider();
        }
```

You can fully get rid of the second interface and all will work the exact same way ;-)

The difference is only how you add the dependecies in this ServiceProvider function. One is added as a transient (so a new one everytime), the second one as a singleton (so unique).

give it a try ;-)

Reply:
> And there is no way to do it in DI, because if you register same interface two times only the second one will be resolved

ho, I see now. So you're most likely right then!

Reply:
![image](https://user-images.githubusercontent.com/19893965/187772076-47757b0c-4175-4dbf-b9af-c580701331f1.png)
Checked that, last one is resolved. So 2 interfaces and 1 class :)